boot/bootutil/include/bootutil/bootutil.h

```diff
@@ -65,6 +65,38 @@ int boot_swap_type(void);
 int boot_set_pending(int permanent);
 int boot_set_confirmed(void);
 
+#define MCUBOOT_REQ_FLASH_MAP_SIZE          0
+#define MCUBOOT_REQ_FLASH_MAP_INFO          1
+
+struct mcuboot_api_itf {
+    uint32_t mcuboot_api_magic;
+    uint32_t mcuboot_version;
+    int (*mcuboot_ioctl)(int req, void *data, size_t len);
+};
+
+extern const struct mcuboot_api_itf mcuboot_api_vt;
+extern struct mcuboot_api_itf *p_mcuboot_api_vt;
+
+#if defined(__arm__)
+    #define MCUBOOT_API_INIT_(m)                                            \
+        extern const struct mcuboot_api_itf mcuboot_api_vt;                 \
+        asm("mov r4, %0\n\t"                                                \
+            "mov r5, %1"  :: "I" (m), "r" (&mcuboot_api_vt) : "r4", "r5");
```

Comment:
Adding the clobbers there means the registers will not be reused for any "asm" stuff apart from the explicit usage: https://gcc.gnu.org/onlinedocs/gcc/Extended-Asm.html#Clobbers. Not explicitly stated but it would not make much sense IMO if they were not preserved.

According to http://infocenter.arm.com/help/topic/com.arm.doc.ihi0042f/IHI0042F_aapcs.pdf r0-r3 are used for parameters and r4-r7 for local variables inside the called function.

Btw, the asm block could be updated to just jump/branch directly instead of having to call the function in C. What do you think?